### PR TITLE
apps/speed: fix possible OOB access in some EC arrays (1.1.0)

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
Cherry-picked from commit 5c6a69f539.

Partial Back-port of #6133 to 1.1.0

##### Checklist

- [X] documentation is added or updated
- [X] tests are added or updated

### CODE REVIEW

**Comment by FdaSilvaYY** [comment]
As #6133 don't cherry-pick cleanly, I have to do some adjustments ;)

**Comment by FdaSilvaYY** [comment]
ping @dot-asm , @richsalz :)
Red cross from Travis CI is unrelated : `/usr/bin/ld: unrecognized option '--push-state--no-as-needed`

File: apps/speed.c [hunk 1]
```diff
@@ -586,7 +552,40 @@ static OPT_PAIR ecdh_choices[] = {
 # define COUNT(d) (count)
 #endif                         /* SIGALRM */
 
-static int testnum;
+static unsigned int testnum;
+typedef void *(*kdf_fn) (
+        const void *in, size_t inlen, void *out, size_t *xoutlen);
```
Comment (dot-asm) [inline apps/speed.c]: While we are at it, I'd suggest to fix formatting as
```
typedef void *(*kdf_fn)(const void *in, size_t inlen, void *out,
                        size_t *xoutlen);
```

**Comment by dot-asm** [comment]
Merged. Thanks!

**Pull request closed without merging**